luarules/gadgets/unit_area_timed_damage.lua

```diff
@@ -15,492 +15,272 @@ if not gadgetHandler:IsSyncedCode() then
- 'scavengerbossv4_veryeasy_scav_turbo_napalm', 'scavengerbossv4_easy_scav_turbo_napalm', 'scavengerbossv4_normal_scav_turbo_napalm', 'scavengerbossv4_hard_scav_turbo_napalm', 'scavengerbossv4_veryhard_scav_turbo_napalm', 'scavengerbossv4_epic_scav_turbo_napalm'}
-for _, name in pairs(scavengerBossV4Table) do
-	TimedDamageWeaponsNames[name] = table.copy(TimedDamageWeaponsNames['scavengerbossv4_normal_turbo_napalm'])
+local frameInterval = math.round(Game.gameSpeed * damageInterval)
+local frameCegShift = math.round(Game.gameSpeed * damageInterval * 0.5)
+
+local timedDamageWeapons = {}
+local unitDamageImmunity = {}
+
+local aliveExplosions = {}
+local frameExplosions = {}
+local frameNumber = 0
+
+local explosionCount = 0
+
+--------------------------------------------------------------------------------
+-- Local functions -------------------------------------------------------------
+
+local function getExplosionParams(def, prefix)
+    local params = {
+        ceg        = def.customParams[ prefix.."ceg"        ],
+        damageCeg  = def.customParams[ prefix.."damageceg"  ],
+        resistance = def.customParams[ prefix.."resistance" ] or resistance,
+        damage     = def.customParams[ prefix.."damage"     ] or damage,
+        frames     = def.customParams[ prefix.."time"       ] or time,
+        range      = def.customParams[ prefix.."range"      ] or range,
+    }
+    params.damage = tonumber(params.damage) * (frameInterval/Game.gameSpeed)
+    params.frames = tonumber(params.frames) * Game.gameSpeed
+    params.frames = math.round(params.frames / frameInterval) * frameInterval
+    params.range = tonumber(params.range)
+    params.resistance = string.lower(params.resistance)
+    return params
 end
 
--- convert weaponname -> weaponDefID
-local TimedDamageWeapons = {}
-for name, params in pairs(TimedDamageWeaponsNames) do
-	if WeaponDefNames[name] then
-		TimedDamageWeapons[WeaponDefNames[name].id] = params
-	end
+-- Change (eg) fire-area-150-repeating to fire-area-<range>-repeating for tweakdefs:
+local sub1, sub2 = '-area-', '-repeating'
+local pattern = sub1..'\d+'..sub2
+local midX, midZ = Game.mapSizeX / 2, Game.mapSizeZ / 2
+local lowY = Spring.GetGroundHeight(midX, midZ) - 10000
+local function getNearestCEG(params)
+    local ceg, range = params.ceg, params.range
+    local sizeBest, diffBest = math.huge, math.huge
+    for ii = 1, #areaSizePresets do
+        local size = areaSizePresets[ii]
+        local diff = math.abs(range / size - size / range)
+        if diff < diffBest then
+            local cegTest = string.gsub(ceg, pattern, sub1..math.floor(sizeBest)..sub2)
+            local success, cegID = Spring.SpawnCEG(cegTest, midX, lowY, midZ) -- hidden-ish
+            if success and cegID then
+                diffBest = diff
+                sizeBest = size
+            end
```

Comment:
The Lua language server doesn't like the `zd+` being wrapped in single quotes `''` instead of double square brackets `[[]]`, so `..[[\d+]]..`. This part is not a blocker

This does not seem to actually parse and correct CEG's to nearest valid ones?
Area of effect seem to be names `-repeat`, not `-repeating-` but this could maybe also do with better error handling.

Reply:
Thanks, I don't know when I switched that over.